dwave/plugins/sklearn/transformers.py

```diff
@@ -0,0 +1,293 @@
+            )
+
+        if num_features <= 0:
+            raise ValueError(f"num_features must be a positive integer, given {num_features}")
+
+        self._alpha = alpha
+        self._method = method
+        self._num_features = num_features
+        self._time_limit = time_limit  # check this lazily
+
+    def __sklearn_is_fitted__(self) -> bool:
+        # used by `check_is_fitted()`
+        try:
+            self._mask
+        except AttributeError:
+            return False
+
+        return True
+
+    def _get_support_mask(self) -> np.ndarray[typing.Any, np.dtype[np.bool_]]:
+        """Get the boolean mask indicating which features are selected
+
+        Returns:
+          boolean array of shape [# input features]. An element is True iff its
+          corresponding feature is selected for retention.
+
+        Raises:
+            RuntimeError: This method will raise an error if it is run before `fit`
+        """
+        check_is_fitted(self)
+
+        try:
+            return self._mask
+        except AttributeError:
+            raise RuntimeError("fit hasn't been run yet")
+
+    def correlation_cqm(
+        self,
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        alpha: float,
+        num_features: int,
+        strict: bool = True,
+    ) -> dimod.ConstrainedQuadraticModel:
+        """Build a constrained quadratic model for feature selection.
+
+        This method is based on maximizing influence and independence as
+        measured by correlation [Milne et al.]_.
+
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.  `alpha=1` places all weight on
+                relevance and selects all features, whereas `alpha=0` places all
+                weight on redundancy and selects no features.
+            num_features:
+                The number of features to select.
+            strict:
+                If ``False`` the constraint on the number of selected features
+                is ``<=`` rather than ``==``.
+
+        Returns:
+            A constrained quadratic model.
+
+        .. [Milne et al.] Milne, Andrew, Maxwell Rounds, and Phil Goddard. 2017. "Optimal Feature
+            Selection in Credit Scoring and Classification Using a Quantum Annealer."
+            1QBit; White Paper.
+            https://1qbit.com/whitepaper/optimal-feature-selection-in-credit-scoring-classification-using-quantum-annealer
+        """
+
+        X = np.atleast_2d(np.asarray(X))
+        y = np.asarray(y)
+
+        if X.ndim != 2:
+            raise ValueError("X must be a 2-dimensional array-like")
+
+        if y.ndim != 1:
+            raise ValueError("y must be a 1-dimensional array-like")
+
```

Comment:
This is a stupid edge case so feel free to ignore, the only likely place it might happen is in testing and I'm raising it just because the resulting error message is confusing: we're not checking X.shape[0], y.shape[0] and if they're 1 for testing, you get an error ``ValueError: y must be a (1, 1) array`` that contradicts the requirement that ``y`` is a 1D array.

Reply:
Minimum failing example by popular request: 

```
data = np.random.uniform(-10, 10, size=(1, 3))
outcome = np.array([int(i) for i in (np.random.uniform(0, 1, size=(1, 1)) > .5)])
selector.fit_transform(data, outcome, num_features=2)
.
ValueError                                Traceback

ValueError: out must be a (1, 1) array
```

Reply:
I see, right, we can add a check for that.